### PR TITLE
Adds checks for removal of ORFs during scoring.

### PR DESCRIPTION
**Pull request by samhorsfield96**
Addresses out of range error, where occasionally multithreading will cause the same item to be remove from a map twice.

### CODE REVIEW

File: src/graph.cpp [hunk 1]
```diff
@@ -399,7 +399,15 @@ std::pair<ColourORFMap, ColourEdgeMap> Graph::findGenes (const bool repeat,
...
                 // get centroid info
                 auto& centroid_ID_pair = ORF_entries.at(0);
-                auto& centroid_info = colour_ORF_vec_map.at(centroid_ID_pair.first).at(centroid_ID_pair.second);
+
+                // ensure centroid present in colour_ORF_vec_map
+                auto centroid_entry = colour_ORF_vec_map[centroid_ID_pair.first].find(centroid_ID_pair.second);
+                if (centroid_entry == colour_ORF_vec_map[centroid_ID_pair.first].end())
+                {
+                    continue;
+                }
```
Comment (johnlees) [inline src/graph.cpp:408]: It might be better to refactor this whole thing as an if/else rather than using continue (which can be a bit hard to follow). But doesn't matter too much – if that's difficult because it's a long or complex loop this is fine

**Review by johnlees — APPROVED** [review]

File: src/graph.cpp (hunk 1 above)
Comment (samhorsfield96) [reply src/graph.cpp:408]: It's because the loop is parallelised, I'm not sure an if/else block would work with how openmp wants for loops to be structured.

**Comment by samhorsfield96** [comment]
> The OS X tests are timing out here:
> 
> ```
> Fri, 24 Nov 2023 12:05:52 GMT 100%|██████████| 1/1 [00:00<00:00, 31300.78it/s]
> Fri, 24 Nov 2023 12:05:55 GMT trimming contig ends...
> Fri, 24 Nov 2023 12:05:55 GMT refinding genes...
> Fri, 24 Nov 2023 12:05:55 GMT Number of searches to perform:  41
> Fri, 24 Nov 2023 12:05:55 GMT Searching...
> Fri, 24 Nov 2023 17:53:05 GMT Error: The operation was canceled.
> ```

I've worked out why this is happening - the OSX implementation doesn't work with the python shared memory objects being used. I'm planning to rewrite these sections fully in C++ to avoid this issue entirely.
